Sources/SpeziSecureStorage/Keys/KeyStorage.swift

```diff
@@ -0,0 +1,130 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import CryptoKit
+import Foundation
+import Spezi
+
+/// A pair of SecKey containing a private and public key.
+public typealias SecKeyPair = (privateKey: SecKey, publicKey: SecKey)
+
+public final class KeyStorage: Module, DefaultInitializable, EnvironmentAccessible, Sendable {
+    public required init() {}
+    
+    // MARK: - Key Handling
+    
+    /// Create a `ECSECPrimeRandom` key for a specified size.
+    /// - Parameters:
+    ///   - tag: The tag used to identify the key in the keychain or the secure enclave.
+    ///   - size: The size of the key in bits. The default value is 256 bits.
+    ///   - storageScope: The  ``SecureStorageScope`` used to store the newly generate key.
+    /// - Returns: Returns the `SecKey` private key generated and stored in the keychain or the secure enclave.
+    @discardableResult
```

Comment:
Shouldn't be needed, it's quite an expensive operation and we should at least warn the clients to use the result 😅

Reply:
I think it would be fine to allow this result be be discardable; given the fact that the key generation already takes a decent amount of time this might not make a huge difference. I don't think we require the user to store or process the private key as they could always get it again if needed.

Reply:
It may not be obvious, but both on Android and on iOS the key is actually already stored in the create function. So, even if you throw away the result, it wouldn't really matter much... I'm ultimately indifferent about this being marked discardableResult or not though, so I would rather keep it as it was.

Reply:
Sounds good; we can keep it as it was for now, makes this backward compatible as well 👍